block-producer/src/batch_builder/batch.rs

```diff
@@ -33,6 +34,7 @@ impl TransactionBatch {
     /// - The number of created notes across all transactions exceeds 4096.
     ///
     /// TODO: enforce limit on the number of created nullifiers.
+    #[instrument(target = "miden-block-producer", skip_all, err)]
     pub fn new(txs: Vec<SharedProvenTx>) -> Result<Self, BuildBatchError> {
         let updated_accounts = txs
             .iter()
```

Comment:
As a part of this constructor, I think we should also compute "batch ID". This would be a new field that we could add to the `TransactionBatch` struct.

A simple way to define a unique ID per batch is to compute a hash of all transaction IDs. For this, we could use some fast hash function. Probably the best option is [Blake3](https://github.com/0xPolygonMiden/crypto/blob/main/src/hash/blake/mod.rs#L107) which we already have in `miden-crypto`.

Lastly, I would probably change the name of the span to `new_batch`.